### PR TITLE
[4.3] Pre-select category for new custom field

### PR DESCRIPTION
**Pull request by joomdonation**
Pull Request for Issue #38794.

### Summary of Changes
This PR implement feature request described https://github.com/joomla/joomla-cms/issues/38794 . Please look at that linked issue to understand the request.

### Testing Instructions
1. Go to Content -> Fields to manage custom fields for articles
2. Select a category from Categories filter option.
3. Press New button in the toolbar to create new custom field

### Actual result BEFORE applying this Pull Request
On new custom field screen, the pre-selected category above is not selected. Category of the field is set to **All**

### Expected result AFTER applying this Pull Request
On new custom field screen, the pre-selected category above is selected. Category of the field is set to the category which you selected in the filter before pressing New button

### CODE REVIEW

**Comment by brianteeman** [comment]
Shouldn't this be applied to all filters not just category?

**Comment by joomdonation** [comment]
@brianteeman Other filters already worked before this PR. For some reasons, category is the only field not pre-selected base on filter, and this PR just improves it.

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 5528470901bebeb82f37278c310890409c245691<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39253">issues.joomla.org/tracker/joomla-cms/39253</a>.</sub>

File: administrator/components/com_fields/src/Model/FieldModel.php [hunk 1]
```diff
@@ -896,6 +896,11 @@ protected function loadFormData()
                 $data->set('state', $input->getInt('state', ((isset($filters['state']) && $filters['state'] !== '') ? $filters['state'] : null)));
                 $data->set('language', $input->getString('language', (!empty($filters['language']) ? $filters['language'] : null)));
                 $data->set('group_id', $input->getString('group_id', (!empty($filters['group_id']) ? $filters['group_id'] : null)));
+                $data->set(
+                    'assigned_cat_ids',
+                    $app->input->get('assigned_cat_ids',
```
Comment (laoneo) [inline administrator/components/com_fields/src/Model/FieldModel.php]: You can use here $input directly.

Comment (joomdonation) [reply administrator/components/com_fields/src/Model/FieldModel.php]: Yes, made the change. Thanks @laoneo.

**Comment by joomdonation** [comment]
As it is a simple and obvious change, I restored @brianteeman test result so that he does not need to test again.

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on 75967e287c2a57c85562531b97aaf71d107d0021<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39253">issues.joomla.org/tracker/joomla-cms/39253</a>.</sub>

**Comment by joomdonation** [comment]
RTC. Thanks @brianteeman and @viocassel for testing. <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39253">issues.joomla.org/tracker/joomla-cms/39253</a>.</sub>

**Pull request merged**